src/Sarif/Baseline/ResultMatching/SarifLogMatcher.cs

```diff
@@ -252,7 +252,8 @@ private Rule GetRuleFromResources(Result result, IDictionary<string, Rule> rules
                 Result result = resultPair.CalculateBasedlinedResult(PropertyBagMergeBehavior);
 
                 IList<FileData> files = 
-                    (PropertyBagMergeBehavior.HasFlag(DictionaryMergeBehavior.InitializeFromOldest) && result.BaselineState == BaselineState.Existing)
+                    (PropertyBagMergeBehavior.HasFlag(DictionaryMergeBehavior.InitializeFromOldest) &&
+                    (result.BaselineState == BaselineState.Unchanged || result.BaselineState == BaselineState.Updated)) 
```

Comment:
>Updated [](start = 110, length = 7)

Need a unit test that shows updated files are included.

Reply:
Not yet! No matcher actually sets that enum value (to indicate, yes, found a match but details were update). That's a good improvement for the heuristics manager to set, potentially. All this area needs a significant update, however, which Scott will do in the future. So deferring for now.

---
In reply to: [252002401](https://github.com/Microsoft/sarif-sdk/pull/1241#discussion_r252002401) [](ancestors = 252002401)